plutus-wallet-api/ledger/Ledger/Index.hs

```diff
@@ -235,16 +236,16 @@ matchInputOutput txid mp i txo = case (txInType i, txOutType txo) of
 --   correct and script evaluation has to terminate successfully. If this is a
 --   pay-to-pubkey output then the signature needs to match the public key that
 --   locks it.
-checkMatch :: ValidationMonad m => (PendingTx, [DataScript]) -> InOutMatch -> m ()
+checkMatch :: ValidationMonad m => (PendingTxPre, [DataScript]) -> InOutMatch -> m ()
 checkMatch (pendingTx, dataScripts) = \case
     ScriptMatch txin vl r d a
         | a /= scriptAddress vl ->
                 throwError $ InvalidScriptHash d
         | otherwise -> do
-            pTxIn <- mkIn txin
-            let vd = ValidationData
-                    $ lifted
-                    $ pendingTx { pendingTxIn = pTxIn }
+            pTxIn <- pendingTxInScript (txInRef txin) vl r
+            let
+                ptx' = fmap (const pTxIn) pendingTx
```

Comment:
This line was quite baffling for a while. I would prefer record update syntax, if possible. If that doesn't work when you change the type, maybe a lens?